### ISSUE TITLE
Issue on custom run action in a specific project (Maven on MAC)

### ISSUE THREAD

**Issue by fabio-franco** [OpenLiberty/liberty-tools-intellij#230]
That's an interesting behavior. I was using the guide-getting-started (https://github.com/OpenLiberty/guide-getting-started) for several tests, running on IntelliJ 2022.2.3 and it was fine. 

![image](https://user-images.githubusercontent.com/7606936/207900495-6c19719f-0509-4b67-ac52-c5bf8815c1e7.png)

However, I installed the latest plugin to check and suddenly I started to face an error on the custom run action. I recorded a video about it:

https://user-images.githubusercontent.com/7606936/207899013-65fae687-977d-4b06-97a8-2d51f12edb1c.mov

In the video, you can see the "guide-getting-started-new" that is a very fresh clone I did to double check the behavior. For this "guide-getting-started-new", the custom run action works fine. However, using the previous project that I used multiple times during the test, the second one that appears in the video ("guide-getting-started"), I can reproduce the error consistently. 

Here's a link with the zip of the project where the issue can be reproduced and with the plugin used: https://ibm.box.com/s/tkqws363ypy4v8lf76fhfs21a661j40z

**Comment by kathrynkodama** [comment]
@fabio-franco Does refreshing the Liberty tool window help at all? After you open the "old" getting-started project.

**Comment by fabio-franco** [comment]
> @fabio-franco Does refreshing the Liberty tool window help at all? After you open the "old" getting-started project.

I just clicked on the refresh icon, ![image](https://user-images.githubusercontent.com/7606936/207903688-bddc7b73-b94d-4cd3-a10c-b21fab24e0cc.png) and then when clicking again on the custom run I got this issue:

```
unLibertyDevTask#update@toolbar (io.openliberty.tools.intellij.actions.RunLibertyDevTask), actionId=io.openliberty.tools.intellij.actions.RunLibertyDevTask, text='Liberty: Run Liberty action'

java.lang.NullPointerException: Parameter specified as non-null is null: method com.intellij.openapi.wm.ToolWindowManager$Companion.getInstance, parameter project
	at com.intellij.openapi.wm.ToolWindowManager$Companion.getInstance(ToolWindowManager.kt)
	at com.intellij.openapi.wm.ToolWindowManager.getInstance(ToolWindowManager.kt)
	at io.openliberty.tools.intellij.actions.RunLibertyDevTask.update(RunLibertyDevTask.java:45)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.lambda$performDumbAwareUpdate$0(ActionUtil.java:150)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareUpdate(ActionUtil.java:173)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doUpdate(ActionUpdater.java:660)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$updateActionReal$4(ActionUpdater.java:128)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$computeOnEdt$6(ActionUpdater.java:211)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:66)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:188)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:175)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:66)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$computeOnEdt$7(ActionUpdater.java:207)
	at com.intellij.openapi.actionSystem.impl.ActionUpdateEdtExecutor.lambda$computeOnEdt$0(ActionUpdateEdtExecutor.java:45)
	at com.intellij.openapi.actionSystem.impl.Utils.lambda$expandActionGroupFastTrack$5(Utils.java:293)
	at com.intellij.openapi.actionSystem.impl.Utils.runLoopAndWaitForFuture(Utils.java:778)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroupFastTrack(Utils.java:291)
	at com.intellij.openapi.actionSystem.impl.ActionToolbarImpl.updateActionsImplFastTrack(ActionToolbarImpl.java:1234)
	at com.intellij.openapi.actionSystem.impl.ActionToolbarImpl.updateActionsImpl(ActionToolbarImpl.java:1204)
	at com.intellij.openapi.actionSystem.impl.ActionToolbarImpl$1.updateActionsImpl(ActionToolbarImpl.java:198)
	at com.intellij.openapi.actionSystem.impl.ToolbarUpdater$MyUpdateRunnable.run(ToolbarUpdater.java:152)
	at com.intellij.openapi.actionSystem.impl.ToolbarUpdater.updateActions(ToolbarUpdater.java:68)
	at com.intellij.openapi.actionSystem.impl.ActionToolbarImpl.updateActionsImmediately(ActionToolbarImpl.java:1183)
	at com.intellij.openapi.actionSystem.impl.ActionToolbarImpl.updateActionsImmediately(ActionToolbarImpl.java:1168)
	at com.intellij.openapi.actionSystem.impl.ActionToolbarImpl.addNotify(ActionToolbarImpl.java:254)
	at java.desktop/java.awt.Container.addImpl(Container.java:1150)
	at java.desktop/java.awt.Container.add(Container.java:1001)
	at com.intellij.openapi.ui.SimpleToolWindowPanel.setToolbar(SimpleToolWindowPanel.java:102)
	at io.openliberty.tools.intellij.actions.RefreshLibertyToolbar.actionPerformed(RefreshLibertyToolbar.java:75)
	at com.intellij.openapi.actionSystem.impl.ActionButton.actionPerformed(ActionButton.java:190)
	at com.intellij.openapi.actionSystem.impl.ActionButton.lambda$performAction$0(ActionButton.java:166)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareWithCallbacks(ActionUtil.java:337)
	at com.intellij.openapi.actionSystem.impl.ActionButton.performAction(ActionButton.java:166)
	at com.intellij.openapi.actionSystem.impl.ActionButton.processMouseEvent(ActionButton.java:465)
	at java.desktop/java.awt.Component.processEvent(Component.java:6413)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5022)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2802)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:754)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:752)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:751)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:918)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:840)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:763)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:450)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:791)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:449)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:624)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:447)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:881)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:493)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
```

After that, I did a invalid cache action: 
![image](https://user-images.githubusercontent.com/7606936/207904271-c3be7e6d-5356-4b91-bba7-b3cf8a603144.png)

and then when I reopened and tried to run the custom action, I got this:
```
java.lang.NullPointerException: Cannot invoke "Object.equals(Object)" because "vBuildFile" is null
	at io.openliberty.tools.intellij.actions.LibertyDevCustomStartAction.lambda$executeLibertyAction$0(LibertyDevCustomStartAction.java:47)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at io.openliberty.tools.intellij.actions.LibertyDevCustomStartAction.executeLibertyAction(LibertyDevCustomStartAction.java:43)
	at io.openliberty.tools.intellij.actions.LibertyGeneralAction.actionPerformed(LibertyGeneralAction.java:123)
	at io.openliberty.tools.intellij.LibertyExplorer.executeAction(LibertyExplorer.java:331)
	at io.openliberty.tools.intellij.LibertyExplorer$2.onDoubleClick(LibertyExplorer.java:259)
	at com.intellij.ui.DoubleClickListener.onClick(DoubleClickListener.java:30)
	at com.intellij.ui.ClickListener$1.mouseReleased(ClickListener.java:59)
	at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:298)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6648)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3392)
	at com.intellij.ui.treeStructure.Tree.processMouseEvent(Tree.java:422)
	at java.desktop/java.awt.Component.processEvent(Component.java:6413)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5022)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2802)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4854)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:754)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:752)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:751)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:918)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:840)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:763)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:450)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:791)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:449)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:624)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:447)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:881)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:493)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

```

**Comment by kathrynkodama** [comment]
Thanks for trying it out, the only other workaround I can think of for now is to close IntelliJ completely and then re-open to your desired project. 

We will have to look at cleaning up references to projects that have already been closed.

**Comment by fabio-franco** [comment]
> Thanks for trying it out, the only other workaround I can think of for now is to close IntelliJ completely and then re-open to your desired project.
> 
> We will have to look at cleaning up references to projects that have already been closed.

Ok. np. Just tried but still failing. I closed the project and then I closed IntelliJ completely. So, I reopened IntelliJ and then opened the project. Still with issue: `java.lang.NullPointerException: Cannot invoke "Object.equals(Object)" because "vBuildFile" is null`.

**Comment by turkeylurkey** [comment]
I could not reproduce the result. I cloned that project and ran the server twice as shown in the video. Then I cloned the same project into a different location and loaded that project in the same IntelliJ window as shown in the video and started the servers again using both Start and Start... but I could not reproduce the exceptions. 

I notice in the video he killed the servers using ^C and maybe that can lead to data errors in the project but I could not reproduce the exact moment he killed the servers. Also he started with one version of the plugin and upgraded to a different version but it is not clear what versions were involved. Possibly they were development versions. 

I note that there is a project supplied and it is possible that the project is corrupt but I don't think we can support a project that is corrupted during development. If we could corrupt a project using just supported work flows we could investigate that but the attempt described in the first paragraph could not. 

@TrevCraw can we remove the must-fix label or close this issue?
